packages/nx/src/command-line/dep-graph.ts

```diff
@@ -398,6 +486,16 @@ async function startServer(
 
     if (sanitizePath === 'project-graph.json') {
       res.writeHead(200, { 'Content-Type': 'application/json' });
+      if (watchForchanges) {
+        recalculateBaseAndHead(affectedWatchArgs);
+        const affected = await recalculateAffectedProjects(affectedWatchArgs);
```

Comment:
I think we should either reuse the entirety of `splitArgsIntoNxArgsAndOverrides` here. Or at least isolate the part which identifies the files based off of the options.